go-controller/pkg/ovn/gateway.go

```diff
@@ -1167,22 +1187,24 @@ func (gw *GatewayManager) syncGatewayLogicalNetwork(
 		return fmt.Errorf("failed to init gateway for network %q: %v", gw.netInfo.GetNetworkName(), err)
 	}
 
-	for _, subnet := range hostSubnets {
-		hostIfAddr := util.GetNodeManagementIfAddr(subnet)
-		if hostIfAddr == nil {
-			return fmt.Errorf("host interface address not found for subnet %q on network %q", subnet, gw.netInfo.GetNetworkName())
-		}
-		l3GatewayConfigIP, err := util.MatchFirstIPNetFamily(utilnet.IsIPv6(hostIfAddr.IP), l3GatewayConfig.IPAddresses)
-		if err != nil {
-			return fmt.Errorf("failed to extract the gateway IP addr for network %q: %v", gw.netInfo.GetNetworkName(), err)
-		}
-		relevantHostIPs, err := util.MatchAllIPStringFamily(utilnet.IsIPv6(hostIfAddr.IP), hostAddrs)
-		if err != nil && err != util.ErrorNoIP {
-			return fmt.Errorf("failed to extract the host IP addrs for network %q: %v", gw.netInfo.GetNetworkName(), err)
-		}
-		pbrMngr := gatewayrouter.NewPolicyBasedRoutesManager(gw.nbClient, gw.clusterRouterName, gw.netInfo)
-		if err := pbrMngr.Add(node.Name, hostIfAddr.IP.String(), l3GatewayConfigIP, relevantHostIPs); err != nil {
-			return fmt.Errorf("failed to configure the policy based routes for network %q: %v", gw.netInfo.GetNetworkName(), err)
+	if gw.clusterRouterName != "" {
```

Comment:
I think you still need this for layer 2. This is for if a pod is trying to reach the node IP. We reroute it to the management port so it can get to the node. Otherwise in shared gateway mode the GR will just consume the packet. You just need to put these on the GR.

Reply:
this is correct. I dont think we need hte masquerade IP either, but it doesn't hurt for now. We can clean that up later.

Reply:
Oh, I had already removed it when I read your reply.

It was pushed as a separate commit; I can get rid of it easily if needed.

Reply:
all good lets keep it

Reply:
https://github.com/ovn-org/ovn-kubernetes/issues/4715